kuma/javascript/src/header/dropdown.test.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { act, create } from 'react-test-renderer';
+import Dropdown from './dropdown.jsx';
+
+test('Dropdown closed and open snapshots', () => {
+    const dropdown = create(
+        <Dropdown label="Test">
+            <li>foo</li>
+            <li>bar</li>
+        </Dropdown>
+    );
+
+    let closedTree = dropdown.toJSON();
+    let closedString = JSON.stringify(closedTree);
+
+    // Expect the menu to be closed and take a snapshot
+    expect(closedString).not.toContain('foo');
+    expect(closedString).not.toContain('bar');
+    expect(closedTree).toMatchSnapshot();
+
+    // Fake a click on the menu label
+    act(() => {
+        dropdown.toJSON().children[0].props.onClick();
+    });
+    let openTree = dropdown.toJSON();
+    let openString = JSON.stringify(openTree);
+
+    // Now expect the menu to be open, and snapshot it in that state
+    expect(openString).toContain('foo');
+    expect(openString).toContain('bar');
+    expect(openTree).toMatchSnapshot();
+
+    // Fake a click on the document body to dismiss the menu
+    act(() => {
+        document.body.dispatchEvent(new MouseEvent('click'));
+    });
+
+    // Expect the menu to be closed, checking deep equality
+    expect(JSON.stringify(dropdown.toJSON())).toEqual(closedString);
```

Comment:
I didn't expect that clicking on the document body would dismiss the menu, given setting ``document.documentElement.style.pointerEvents = 'none';`` in ``Dropdown``'s ``useEffect``, but I just learned about [pointer-events](https://developer.mozilla.org/en-US/docs/Web/CSS/pointer-events) a few hours ago, so I must be misunderstanding the purpose of setting ``pointerEvents``.

Reply:
The click and keydown event handlers I register for dismissing the menu are capturing handlers, which means they are triggered before the event gets dispatched anywhere (during the "capturing phase", before the event is dispatched to its target and then bubbles back up). This means my handlers get first crack at the events.

Setting style.pointerEvents to none on the html element makes everything "transparent" to events. The main point there is to make it so that things with hover behaviors don't do anything when hovered over while the menu is displayed. But it also means that I don't have to call preventDefault() on the event to stop it from actually activating links if the user clicks on one.   (I could instead call preventDefault on events, but would have to be careful to only do that when they were outside of the menu, and it is easier to just do the pointerEvents thing and get the hover disable effect as well.)

Note also that the pointer-events CSS thing is completely different from PointerEvents (which is an attempt to unify mouse and touch events into a single thing, I think.)

Reply:
That helps, thanks @davidflanagan!